### PR TITLE
[1LP][RFR] New Test: Verifying dashboard of Remote and Global Appliances

### PR DESCRIPTION
**Pull request by dgaikwad**
## Purpose or Intent
- Verifying the dashboard of remote appliance and global appliance after adding the vm in the provider.
- Provided one more xpath for dashboard UI(widget table)
- Added one function to compare two dictionaries

### PRT Run

{{pytest: cfme/tests/test_replication.py::test_replication_global_region_dashboard --long-running -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by john-dupuy** [comment]
@dgaikwad please check the 5.10 failure 
```
            )
E           wait_for.TimedOutError: Could not do 'Waiting for data item of EVM: Recently Discovered VMs table.' at /cfme/cfme_tests/cfme/tests/test_replication.py:400 in time

cfme/tests/test_replication.py:402: TimedOutError
TimedOutError
b"Could not do 'Waiting for data item of EVM: Recently Discovered VMs table.' at /cfme/cfme_tests/cfme/tests/test_replication.py:400 in time"```

File: cfme/tests/test_replication.py [hunk 1]
```diff
@@ -342,7 +361,60 @@ def test_replication_global_region_dashboard():
...
+            message=f"Waiting for data item of {table} table."
+        )
+        chart_data = get_tabel_data(remote_widget)
+        remote_app_data[table] = chart_data
+
+    global_view = navigate_to(global_app.server, "Dashboard")
+    global_view.browser.refresh()
+    global_dashboard = global_view.dashboards("Default Dashboard")
+    global_app_data = {}
+    for table in data_items:
```
Comment (ganeshhubale) [inline cfme/tests/test_replication.py]: **global_app_data** is already declared above: ```remote_app_data = global_app_data = {}```

File: cfme/tests/test_replication.py [hunk 2]
```diff
@@ -342,7 +361,60 @@ def test_replication_global_region_dashboard():
...
+        global_app_data[table] = chart_data
+
+    # TODO(ndhandre): Widget not implemented so some widget not checking in this test case
+    #  'Vendor and Guest OS Chart', 'Top Memory Consumers (weekly)', 'Top CPU Consumers (weekly)',
+    #  'Virtual Infrastructure Platforms', 'Guest OS Information'
+
+    assert is_dicts_same(remote_app_data, global_app_data), "Dashboard is not same."
+
+    # CleanUp
+    vm.cleanup_on_provider()
```
Comment (ganeshhubale) [inline cfme/tests/test_replication.py]: Is this teardown?
If yes then please add in addfinalizer above assert.

Comment (dgaikwad) [reply cfme/tests/test_replication.py]: yes, now added in request.addfinalizer()

File: cfme/tests/test_replication.py (hunk 1 above)
Comment (dgaikwad) [reply cfme/tests/test_replication.py]: Done

File: cfme/tests/test_replication.py [hunk 3]
```diff
@@ -29,6 +31,24 @@ def create_vm(provider, vm_name):
     return vm
 
 
+def is_dicts_same(dict1, dict2):
```
Comment (digitronik) [inline cfme/tests/test_replication.py]: As per observation values are list. is it not possible to reduce code with `set` operation?

File: cfme/tests/test_replication.py [hunk 4]
```diff
@@ -342,7 +361,57 @@ def test_replication_global_region_dashboard():
...
+    new_vm_name = fauxfactory.gen_alphanumeric(start="test_rep_dashboard", length=25).lower()
+    vm = create_vm(provider=remote_provider, vm_name=new_vm_name)
+    request.addfinalizer(vm.cleanup_on_provider)
+    data_items = ('EVM: Recently Discovered Hosts', 'EVM: Recently Discovered VMs',
+                  'Top Storage Consumers')
+
+    remote_dashboard = remote_app.collections.dashboards.instantiate(name="Default Dashboard")
+    remote_view = navigate_to(remote_app.server, "Dashboard")
+    remote_view.browser.refresh()
+    remote_dashboard = remote_view.dashboards("Default Dashboard")
```
[the remark this answers is not in the document]
Comment (dgaikwad) [reply cfme/tests/test_replication.py]: anyhow, we have to pass the "Default Dashboard" parameter.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/test_replication.py [hunk 5]
```diff
@@ -29,6 +31,16 @@ def create_vm(provider, vm_name):
     return vm
 
 
+def is_dicts_same(dict1, dict2):
```
Comment (kedark3) [inline cfme/tests/test_replication.py]: grammar nitpick, I think it should be `are_dicts_same` or `dicts_compare`

File: cfme/tests/test_replication.py [hunk 6]
```diff
@@ -329,7 +340,55 @@ def test_replication_global_region_dashboard():
             1.
             2. Dashboard on the Global displays data from the Remote region
     """
-    pass
+    remote_app, global_app = setup_replication
+    remote_provider = provider_app_crud(RHEVMProvider, remote_app)
```
Comment (kedark3) [inline cfme/tests/test_replication.py]: Any reason to choose `RHEVMProvider`? How about simply `InfraProvider` or something of that effect, which will be less tied to one type of provider.

**Comment by dgaikwad** [comment]
@kedark3 I have done the requested changes, please look once.

File: cfme/tests/test_replication.py [hunk 7]
```diff
@@ -329,7 +341,55 @@ def test_replication_global_region_dashboard():
...
+    remote_provider.setup()
+    assert remote_provider.name in remote_app.managed_provider_names, "Provider is not available."
+
+    new_vm_name = fauxfactory.gen_alphanumeric(start="test_rep_dashboard", length=25).lower()
+    vm = create_vm(provider=remote_provider, vm_name=new_vm_name)
+    request.addfinalizer(vm.cleanup_on_provider)
+    data_items = ('EVM: Recently Discovered Hosts', 'EVM: Recently Discovered VMs',
+                  'Top Storage Consumers')
+
+    remote_app_data = global_app_data = {}
```
Comment (kedark3) [inline cfme/tests/test_replication.py]: I think this is dangerous. Do you want both of these dicts to store same values?
Example: 
```
>>> remote_app_data = global_app_data = {} 
>>> remote_app_data['var1'] = 'val1'
>>> remote_app_data
{'var1': 'val1'}
>>> global_app_data
{'var1': 'val1'}

```

As you can see if you modify one dict, the other dict also looks the same. 

If this is undesired, you may use,
```remote_app_data, global_app_data = {},{} ```

Comment (dgaikwad) [reply cfme/tests/test_replication.py]: @kedark3  Thank you for review, will update PR accordingly

**Pull request merged**